pkg/backend/filestate/backend.go

```diff
@@ -546,7 +562,12 @@ func (b *localBackend) apply(
 		} else {
 			link, err = b.bucket.SignedURL(context.TODO(), b.stackPath(stackName), nil)
 			if err != nil {
-				return changes, result.FromError(errors.Wrap(err, "Could not get signed url for stack location"))
+				// we log a warning here rather then returning an error to avoid exiting
+				// pulumi with an error code.
+				// printing a statefile perma link happens after all the providers have finished
+				// deploying the infrastructure, failing the pulumi update because there was a
+				// problem printing a statefile perma link can be missleading in automated CI environments.
```

Comment:
```suggestion
				// problem printing a statefile perma-link can be misleading in automated CI environments.
```

Reply:
```suggestion
				// problem printing a statefile permalink can be misleading in automated CI environments.
```

Reply:
Imho as a user this warning is redundant (for default google `application_default_credentials.json`), since you already saw a [warning](https://github.com/pulumi/pulumi/pull/2906/files#diff-475e825ab79ced972991a5892bdd29e0R72) at login about the need to use service accounts if you want state urls. 
To avoid that, you may return an additional flag from `GoogleCredentialsMux` and use that here to skip the warning, in case the credentials are not based on a service account.

What do you think?